src/server/config.rs

```diff
@@ -91,4 +120,37 @@ mod tests {
         cfg.raft_store.raft_heartbeat_ticks = 0;
         assert!(cfg.validate().is_err());
     }
+
+    #[test]
+    fn test_end_point_txn_concurrency() {
+        let mut cfg = Config::new();
+        let expect = ((cfg.end_point_concurrency as f64) *
+                      DEFAULT_END_POINT_TXN_CONCURRENCY_RATIO) as usize;
+        assert_eq!(cfg.end_point_txn_concurrency_on_busy, expect);
+        cfg.end_point_concurrency = 18;
+        cfg.auto_adjust_end_point_txn_concurrency();
+        let expect = ((cfg.end_point_concurrency as f64) *
+                      DEFAULT_END_POINT_TXN_CONCURRENCY_RATIO) as usize;
+        assert_eq!(cfg.end_point_txn_concurrency_on_busy, expect);
+
+        cfg.end_point_concurrency = 2;
+        cfg.auto_adjust_end_point_txn_concurrency();
+        let expect = 1;
+        assert_eq!(cfg.end_point_txn_concurrency_on_busy, expect);
```

Comment:
Merge line 138, 139 to 
`assert_eq!(cfg.end_point_txn_concurrency_on_busy, 1);`